common/djangoapps/student/models.py

```diff
@@ -1155,7 +1155,7 @@ def is_paid_course(self):
         """
         paid_course = CourseMode.objects.filter(Q(course_id=self.course_id) & Q(mode_slug='honor') &
                                                 (Q(expiration_datetime__isnull=True) | Q(expiration_datetime__gte=datetime.now(pytz.UTC)))).exclude(min_price=0)
-        if paid_course or self.mode == 'professional' or self.mode == CourseMode.NO_ID_PROFESSIONAL_MODE[0]:
+        if paid_course or CourseMode.is_professional_slug(self.mode):
             return True
 
         return False
```

Comment:
is_paid_course() is used on dashboard on registered courses listing. It is used with the combination of show_refund_option flag to show different message about refund if user is unrolling from a paid course. I think professional courses are also paid courses so I included that courses in the list too.